core/controllers/skill_editor.py

```diff
@@ -146,6 +146,50 @@ class EditableSkillDataHandler(base.BaseHandler):
 
     GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
 
+    URL_PATH_ARGS_SCHEMAS = {
+        'skill_id': {
+            'schema': {
+                'type': 'basestring',
+                'validators': [{
+                    'id': 'is_regex_matched',
+                    'regex_pattern': constants.ENTITY_ID_REGEX
+                }]
+            }
+        }
+    }
+
+    HANDLER_ARGS_SCHEMAS = {
+        'GET': {},
+        'PUT': {
+            'version': {
+                'schema': {
+                    'type': 'int',
+                    'validators': [{
+                        'id': 'is_at_least',
+                        'min_value': 1
+                    }]
+                }
+            },
+            'commit_message': {
+                'schema': {
+                    'type': 'basestring'
+                },
+                'default_value': None
+            },
+            'change_dicts': {
+                'schema': {
+                    'type': 'list',
+                    'items': {
+                        'type': 'object_dict',
+                        'object_class': (
+                            skill_domain.SkillChange)
+                    }
+                }
+            }
+        },
+        'DELETE': {}
+    }
+
     @acl_decorators.open_access
     def get(self, skill_id):
         """Populates the data on the individual skill page."""
```

Comment:
The four lines below this line are not needed as we already validate this in the schema.

Reply:
Done!

Reply:
Remove this section also in the `delete` function.

Reply:
Done

Reply:
Also, can be removed from PUT (sorry I missed it before, but also please try to catch these things yourself)

Reply:
Not done.

Reply:
Done